lib/src/react_client/chain_refs.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:react/react_client/react_interop.dart';
+
+/// Returns a ref that updates both [ref1] and [ref2], effectively
+/// allowing you to set multiple refs.
+///
+/// Useful when you're using ref forwarding and want to also set your own ref.
+///
+/// For more information on this problem, see https://github.com/facebook/react/issues/13029.
+///
+/// Inputs can be callback refs, [createRef]-based refs ([Ref]),
+/// or raw JS `createRef` refs ([JsRef]).
+///
+/// If either [ref1] or [ref2] is null, the other ref will be passed through.
+///
+/// If both are null, null is returned.
+///
+/// ```dart
+/// final Example = forwardRef((props, ref) {
+///   final localRef = useRef<FooComponent>();
+///   return Foo({
+///     ...props,
+///     'ref': chainRefs(localRef, ref),
+///   });
+/// });
+/// ```
+dynamic chainRefs(dynamic ref1, dynamic ref2) {
+  return chainRefList([ref1, ref2]);
+}
+
+/// Like [chainRefs], but takes in a list of [refs].
+dynamic chainRefList(List<dynamic> refs) {
+  refs.forEach(_validateChainRefsArg);
```

Comment:
Any reason _(other than the lack of individual error messages)_ to not wrap this whole thing in an assert so the refs are not iterated at all in dart2js?

```dart
dynamic chainRefList(List<dynamic> refs) {
  assert(refs.every(_isChainedRefValid));

  // ...
}

bool _isChainedRefValid(dynamic ref) {
  if (ref is Function) return ref is! Function(Null);
  return ref is! String;
}
```

Reply:
Good point! I'll update this